pkg/volume/iscsi/iscsi_util.go

```diff
@@ -390,3 +407,31 @@ func extractPortalAndIqn(device string) (string, string, error) {
 	iqn := device[ind2:ind]
 	return portal, iqn, nil
 }
+
+func removeDuplicatePortal(bkpPortal []string) []string {
+	portals := make([]string, 0, len(bkpPortal))
+	found := map[string]bool{}
+	for _, tp := range bkpPortal {
+		if !found[tp] {
+			found[tp] = true
+			portals = append(portals, tp)
+		}
+	}
+	return portals
```

Comment:
My go fu is not great but this looks like it will return an array with empty elements at the end in case it removes some duplicates which would mean the kubelet tries to execute iscsiadm commands on empty data.

This go playground example avoids it, maybe can use that
https://play.golang.org/p/q3bZ3hpOzD